gluten-core/src/main/java/io/glutenproject/substrait/expression/ExpressionBuilder.java

```diff
@@ -184,6 +184,11 @@ public static ExpressionNode makeLiteral(Object obj, DataType dataType, Boolean
     } else if (dataType instanceof DecimalType) {
       if (obj == null) {
         DecimalType decimal = (DecimalType)dataType;
+        if (decimal.scale() < 0) {
```

Comment:
This branch is only for null, maybe you could add this check for both null and non-null value.

Reply:
Done